hnn_core/viz.py

```diff
@@ -168,7 +170,8 @@ def plot_laminar_lfp(times, data, contact_labels, tmin=None, tmax=None,
         trace_offsets = np.arange(n_offsets)[:, np.newaxis] * voltage_offset
 
     for contact_no, trace in enumerate(np.atleast_2d(data)):
-        plot_data, plot_times = _get_plot_data_trange(times, trace, tmin, tmax)
+        plot_data = trace
+        plot_times = times
```

Comment:
Not sure if we mean the same thing here. The result will be the same as before: the data will be plotted from tmin to tmax if they are not none. Before, plot_data and plot_times ranged from tmin to tmax, now plot_data and plot_times range from 0 to tstop, but the x-axis will be adjusted to range from tmin to tmax if not None. I have tested this locally and it does not break the scripts. My logic was that this way, we could easily remove the xlim setting after the deprecation cycle (it's commented in the code). 

I also saw that because xlim wasn't set in the function, there was some whitespace on the left and/or right side of the plot when tmin and tmax are set to a value that is not None and sharex=True, as in this example https://jonescompneurolab.github.io/hnn-core/dev/auto_examples/howto/plot_record_extracellular_potentials.html#sphx-glr-auto-examples-howto-plot-record-extracellular-potentials-py

There is no whitespace for this version anymore, see:

![tstop_fix](https://github.com/jonescompneurolab/hnn-core/assets/26790195/b0f3016b-3a2e-44fb-9a56-7c9af3fd5342)


For psd and csd tmin and tmax are still used in _get_data_range following your vote in the first PR. I understand that this will save the user time and compute when calculating CSD and PSD. Hope this is okay. Will address the rest tomorrow with a fresh mind!